src/aspire/nufft/finufft.py

```diff
@@ -10,84 +10,73 @@
-        transforms.  Implies vol_f.shape is (..., `ntransforms`). Defaults to 0 which disables batching.
+        :param sz: A tuple indicating the geometry of the signal.
+        :param fourier_pts: The points in Fourier space where the Fourier
+        transform is to be calculated, arranged as a dimension-by-K array.
+        These need to be in the range [-pi, pi] in each dimension.
+        :param epsilon: The desired precision of the NUFFT.
+        :param ntransforms: Optional integer indicating if you would like
+        to compute a batch of `ntransforms`.
+        transforms.  Implies vol_f.shape is (`ntransforms`, ...).
         """
 
         self.ntransforms = ntransforms
-        manystr = ""
-        if self.ntransforms > 1:
-            manystr = "many"
 
         self.sz = sz
         self.dim = len(sz)
 
         self.dtype = fourier_pts.dtype
 
-        # TODO: Currently/historically finufftpy is hardcoded as doubles only (inside the binding layer).
-        #       This has been changed in their GuruV2 work,
-        #         and an updated package should be released and integrated very soon.
-        #       The following casting business is only to facilitate the transition.
-        #       I don't want to send one precision in and get a different one out.
-        #         We have enough code that does that already.
-        #         (Potentially anything that used this, for example).
-        #       I would error, but I know a bunch of code ASPIRE wants to work,
-        #         the cov2d tutorial for example, would fail and require hacks to run
-        #         if I was strict about dtypes rn.
-        #       I would rather deal with that in other, more targeted PRs.
-        #       This preserves the legacy behavior of admitting singles,
-        #         but I will correct it slightly, to return the precision given as input.
-        #       This approach should contain the hacks here to a single place on the edge,
-        #         instead of spread through the code. ASPIRE code should focus on being
-        #         internally consistent.
-        #       Admittedly not ideal, but ignoring these problems wasn't sustainable.
-
-        self.cast_output = False
-        if self.dtype != np.float64:
-            logger.debug(
-                "This version of finufftpy is hardcoded to doubles internally"
-                "  casting input to doubles, results cast back to singles."
-            )
-            self.cast_output = True
-            self.dtype = np.float64
-
         self.complex_dtype = complex_type(self.dtype)
 
-        # TODO: Things get messed up unless we ensure a 'C' ordering here - investigate why
-        self.fourier_pts = np.asarray(
-            np.mod(fourier_pts + np.pi, 2 * np.pi) - np.pi, order="C", dtype=self.dtype
+        self.fourier_pts = np.ascontiguousarray(
+            np.mod(fourier_pts + np.pi, 2 * np.pi) - np.pi
```

Comment:
Ideally, this modulus should be handled by Finufft. If we're getting errors here, we should report it upstream.